presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerifierConfig.java

```diff
@@ -407,4 +410,21 @@ public VerifierConfig setRunningMode(String runningMode)
         }
         return this;
     }
+
+    public Optional<String> getFunctionSubstitutes()
+    {
+        return functionSubstitutes;
+    }
+
+    @ConfigDescription("Specification of function substitutions, in the format of" +
+            " /foo(c0,_)/bar(c0)/,/fred(c0,c1)/baz(qux(c1,c0))/,/foobar(c0)/if(qux(c1),bar(c0),baz(c1))/,...," +
+            " where foo(c0, _) would be substituted by bar(c0), with the declared arguments applied" +
+            " to the corresponding positions. Concatenate function substitutions with comma.")
+    @Config("function-substitutes")
+    public VerifierConfig setFunctionSubstitutes(String functionSubstitutes)
+    {
+        this.functionSubstitutes = Optional.ofNullable(functionSubstitutes);
+        this.functionSubstitutes.ifPresent(FunctionCallRewriter::validateAndConstructFunctionCallSubstituteMap);
```

Comment:
Since the pattern is parsed and validated here I wonder if it would be cleaner to to set the `functionSubstitutes` type to `Multimap<String, FunctionCallSubstitute>` and return it from the `getFunctionSubstitutes`. The `setFunctionSubstitutes` could still take `String`, it should be fine. With `Multimap` you also don't need optional. You can set it to an empty multimap by default.

Reply:
Yes, I think it works. Revised